graphstorm-processing/graphstorm_processing/repartition_files.py

```diff
@@ -260,31 +309,37 @@ def repartition_parquet_files_in_memory(
         logging.debug("Desired counts: %s", desired_counts)
         logging.debug("Row counts: %s", data_entry_dict["row_counts"])
 
-        offset = 0
-        new_data_entries = []
         # From the dataset we read into memory, we slice a part according to desired_counts and
         # write a new file to S3.
-        for idx, desired_count in enumerate(desired_counts):
-            sliced_data = table.slice(offset=offset, length=desired_count)
-            new_relative_path = self.create_new_relative_path_from_existing(datafile_list[0], idx)
-            self.write_parquet_to_relative_path(new_relative_path, sliced_data, desired_count)
-            new_data_entries.append(new_relative_path)
-            offset += desired_count
+        offsets = accumulate([0] + desired_counts)
+        zero_copy_slices = [
+            table.slice(offset=offset, length=desired_count)
+            for offset, desired_count in zip(offsets, desired_counts)
+        ]
+        uid_for_entry = uuid.uuid4().hex[:8]
+        relative_paths = [
+            self.create_new_relative_path_from_existing(datafile_list[0], idx, uid_for_entry)
+            for idx in range(len(desired_counts))
+        ]
+        with Parallel(n_jobs=min(16, os.cpu_count()), verbose=10, prefer="threads") as parallel:
```

Comment:
what if len(desired_counts) < 16? 
Parallel will automatically handle it?

Reply:
Yeah it will launch as many tasks as needed.